Extension/src/LanguageServer/Providers/codeActionProvider.ts

```diff
@@ -168,9 +168,13 @@ export class CodeActionProvider implements vscode.CodeActionProvider {
                 resultCodeActions.push(...disableCodeActions);
                 resultCodeActions.push(...docCodeActions);
                 return;
-            }
-            if (command.command === 'C_Cpp.CreateDeclarationOrDefinition' && (command.arguments ?? []).length === 0) {
+            } else if (command.command === 'C_Cpp.CreateDeclarationOrDefinition' && (command.arguments ?? []).length === 0) {
                 command.arguments = ['codeAction']; // We report the sender of the command
+            } else if (command.command === "C_Cpp.SelectDefaultCompiler") {
+                const defaultCompilerPath: string | undefined = new CppSettings(this.client.RootUri).defaultCompilerPath;
```

Comment:
I think it would be better to hide the code action if `compilerPath` is set in the active configuration (c_cpp_properties.json) instead.  The user could have selected an unsupported compiler previously, so `C_Cpp.default.compilerPath` could be set, but it's "unsuitable."

Reply:
sure, I'll also looking into https://github.com/microsoft/vscode-cpptools/issues/10674